Makefile

```diff
@@ -2,36 +2,36 @@
 
 # Run PettingZoo example
 pz:
-	python3 -m examples.pettingzoo_example
+	poetry run python3 -m examples.pettingzoo_example
```

Comment:
I think its better its better to have commands that are not under developer to run without poetry, so users don't need to have it installed when just trying to run an example to see what is going on

Reply:
Actually, I think that using `poetry run` here is correct. You need to realize that you are switching from simple requirements (pip) to poetry. If you use plain `python3 ...` it will use system python, not virtual environment one. Which you don't want, because the system python will be different for each user.

True, you force developers of this tool to install `poetry`. On the other hand, users will install and import this tool as library of package and there it does not matter which package manager is used here as they are interacting with build project.

Reply:
I will add the requirement to install poetry to Readme to Installation section.

Reply:
This way, if someone clones the repository, he has to first install poetry in order to run examples right?

The biggest priority is to have as frictionless first interaction with the repository as possible.
Such interaction should look like this:

```python
git clone https://github.com/strakam/generals-bots
cd generals-bots
pip install -e .
make gym
```

If this procedure crashes on `make gym` because `poetry` is not installed, it gives a sense that the repository is not as lightweight as we present it to be.

I don't have experience with `poetry`, but if there is a workaround for this, I'll be happy to hear it.

Reply:
I am afraid not. Poetry replaces pip for dependency management, therefore you must use `poetry install --with dev` instead of `pip install -e .`. 

There are benefits of poetry over pip, some of them are:
- `poetry.lock` ensures that dependencies are installed consistently across environments
- automatically manages venvs
- simplifies package publishing (you don't need a separate tool like `twine`, you just do `poetry publish`)

I think that poetry is now quite known and used across many projects and I think that running extra `pip install poetry` won't decrease the ease of interacting with this project. Pip is older and the default package manager so 100% of people using python know pip. At the same time, the user would need to call one extra command of poetry installation (using pip which developers know) and then `poetry install --with dev` instead of `pip install -e .`. But if you insist on `pip install -e .` then we can discard this PR and maybe return to this in the future:)

Reply:
Taking your example above, with poetry, it would look like this:
```
git clone https://github.com/strakam/generals-bots
cd generals-bots
pip install poetry
poetry install --with dev
make gym
```

Or we could maybe even add a `make install` abstraction, so the user would do:
```
git clone https://github.com/strakam/generals-bots
cd generals-bots
make install
make gym
```
where `make install` runs 
```
pip install poetry
poetry install --with dev
```

Reply:
I like the one with abstraction. I suggest updating `README.md` accordingly (but omit `make gym`).